### ISSUE TITLE
chunksize vs. chunk_size inconsistency coordinates api

### ISSUE THREAD

**Issue by marscher** [markovmodel/PyEMMA#1235]
There is a huge inconsistency within these parameters in the api. Since we have released this BS already, how should we properly unify this without breaking peoples code?

@clonker since you introduced chunk_size, I hope you can provide a solution 👍

**Comment by clonker** [comment]
Hmmm. In my defense there was no api named argument called `chunksize` when i introduced `chunk_size`... however I probably should have stuck to the already present naming conventions in non-api functions/attributes, sorry for that.
As for a solution the best that comes to my mind right now is that (almost) all coordinates api functions have support for additional `**kw` - so we could deprecate one of the two and support both for a while until one version can safely be removed or simply support both indefinitely.

**Comment by marscher** [comment]
Sounds great. I'd like to have that prior 2.5. We just need a function picking out cs from kw and trigger a deprecation warning. 
Shall we use chunksize, which is also the name of the property in DataSource or chunk_size, which seems to be correct in terms of spelling?

**Comment by clonker** [comment]
Since the DataSource is exposed by the api we should go for chunksize - although I would actually prefer chunk_size.

**Comment by marscher** [comment]
On 02/02/2018 03:54 PM, Moritz Hoffmann wrote:
> Since the DataSource is exposed by the api we should go for chunksize -
> although I would actually prefer chunk_size.
I have noticed, that chunk_size has already been released in 2.4 for
most api funcs. We can also revert that. I think most users are not
aware of this parameter, because there much more important things to
think about.

The prepared Pr now switched everything to chunk_size and warns about
chunksize. But to be fully consistent one would have to remove the
DatsSource stuff to (which is much much more work).
